Legacy DB upgrade fixes, refs #13350

jraddaoui opened the pull request:
See [RM#13350](https://projects.artefactual.com/issues/13350) and commit messages for more information.

djjuhasz commented on lib/task/migrate/arUpgrader120.class.php:
```diff
@@ -244,7 +244,7 @@ public function up($version, $configuration, $options)
           foreach ($cols as $col)
           {
-            $sql = sprintf('ALTER TABLE %s MODIFY %s VARCHAR(1024) NOT NULL;', $tablename, $col);
+            $sql = sprintf('ALTER TABLE `%s` MODIFY %s VARCHAR(1024) NOT NULL;', $tablename, $col);
```
Nitpick: I'd add backticks around the column name here too, for consistency

jraddaoui replied:
Lets hope they don't become reserved too :P

djjuhasz commented on lib/task/migrate/migrations/arMigration0098.class.php:
```diff
@@ -36,47 +36,21 @@ class arMigration0098
-    // Attempt to rename old column name to new column name so newer
-    // ORM will work with it
-    try {
-      $sql = "ALTER TABLE `event` CHANGE `information_object_id` `object_id` INT(11) DEFAULT NULL";
```
Why is this no longer required?

jraddaoui replied:
This comes from a system we had to do in times of 1.4. Some of this migrations (3 or 4 only) have a `MIN_MILESTONE = 1` which indicates they are needed for a 1.x to 1.4 upgrade. To be fair, I never understood this process entirely, but you can see the migrations included in qa/1.4.x:

https://github.com/artefactual/atom/tree/qa/1.4.x/lib/task/migrate/migrations

What I noticed now is that, if you're upgrading from 1.x to 2.6, those migrations get executed alongside all the others with `MIN_MILESTONE = 2`. Which causes the modification of this column twice (in here and in arMigration0121) and it's probably the reason of having the try/catch blocks.

Since we never released 1.4, all upgrades from 1.x pass through both migrations and with this changes, apart from avoiding the duplication, we can remove the try/catch as this is an important change and it's probably better to raise the exception.

PS. In the end, I think the error mentioned in the ticket about this migration was a non issue (probably caused by some changes along the process). However, I did all the final tests with this fix and I thought it will be nice to have it for the reasons explained above.

djjuhasz replied:
Ohh, interesting.  I forgot all about the 1.4/2.0 fork. 😮  Thanks for the explanation. :)

djjuhasz reviewed (commented):
@jraddaoui I don't understand why the arMigration0098 doesn't need the code to change the `event.information_object_id` column and foreign key relationship to `event.object_id` 😕

jraddaoui commented:
Thanks for the quick review @djjuhasz!

djjuhasz reviewed (approved):
LGTM! 👍

jraddaoui committed:
Legacy DB upgrade fixes, refs #13350

Squashed commit of the following:

commit e3a03a5fffe7c14bd130f9d6cbdc59283d5a4241
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Fri Jun 12 15:04:03 2020 +0200

    Add backticks around the column name

commit 81305090d71c47331b2cce2d5c6754bc853e024e
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Thu Jun 11 21:27:27 2020 +0200

    Avoid ORM on DO update (migration 70), refs #13350

    QubitDigitalObject::INFORMATION_OBJECT_ID doesn't exist in this version.

commit 8e65e5a391be70886b632298ae2548da9288d406
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Thu Jun 11 21:16:50 2020 +0200

    Fix event foreign key update, refs #13350

    - Avoid double update comming from 1.x.
    - Do not catch errors renaming the column.

commit 7ca241fb31b636eb14119355ab6c525bf0dd6db9
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Thu Jun 11 17:40:57 2020 +0200

    Use raw SQL to create root repository, refs #13350

    Avoids the use of the ORM as it's not in sync. with the current DB
    status, causing an unknown column error:

    Unknown column 'digital_object.OBJECT_ID' in 'field list'.

commit f165670ca21fc3e241b08ef74a6a750f43e7ea8f
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Thu Jun 11 17:14:25 2020 +0200

    Fix alter table statements in upgrade, refs #13220

    The use of `function` as table name makes it fail in the 2.6.x upgrade,
    where `function` is a reserved word in MySQL 8.
